This adds support for the GTA01v4 hardware revision of Neo1973

Signed-off-by: Harald Welte <laforge@openmoko.org>

Index: u-boot.git/Makefile
===================================================================
--- u-boot.git.orig/Makefile	2007-02-06 12:04:53.000000000 +0100
+++ u-boot.git/Makefile	2007-02-06 12:07:48.000000000 +0100
@@ -1907,8 +1907,10 @@
 sbc2410x_config: unconfig
 	@$(MKCONFIG) $(@:_config=) arm arm920t sbc2410x NULL s3c24x0
 
-gta01_config	:	unconfig
-	@./mkconfig $(@:_config=) arm arm920t neo1973 NULL s3c24x0
+gta01_config \
+gta01v3_config \
+gta01v4_config :	unconfig
+	@board/neo1973/split_by_variant.sh $@
 
 qt2410_config	:	unconfig
 	@./mkconfig $(@:_config=) arm arm920t qt2410 NULL s3c24x0
Index: u-boot.git/board/neo1973/neo1973.c
===================================================================
--- u-boot.git.orig/board/neo1973/neo1973.c	2007-02-06 12:04:53.000000000 +0100
+++ u-boot.git/board/neo1973/neo1973.c	2007-02-06 12:07:48.000000000 +0100
@@ -91,6 +91,7 @@
 	delay (8000);
 
 	/* set up the I/O ports */
+#if defined(CONFIG_ARCH_GTA01_v3)
 	gpio->GPACON = 0x007FFFFF;
 
 	gpio->GPBCON = 0x00005056;
@@ -113,6 +114,34 @@
 
 	gpio->GPHCON = 0x0008FAAA;
 	gpio->GPHUP = 0x000007FF;
+#elif defined(CONFIG_ARCH_GTA01_v4)
+	gpio->GPACON = 0x005E47FF;
+
+	gpio->GPBCON = 0x00045016;
+	gpio->GPBUP = 0x000007FF;
+	gpio->GPBDAT |= 0x4;		/* Set GBP2 to high (Flash power-up) */
+
+	gpio->GPCCON = 0xAAAA12A9;
+	gpio->GPCUP = 0x0000FFFF;
+
+	gpio->GPDCON = 0xAAAAAAAA;
+	gpio->GPDUP = 0x0000FFFF;
+
+	gpio->GPECON = 0xA02AAAAA;
+	gpio->GPEUP = 0x0000FFFF;
+
+	gpio->GPFCON = 0x0000aa09;
+	gpio->GPFUP = 0x000000FF;
+
+	gpio->GPGCON = 0xFF40F0C1;
+	gpio->GPGUP = 0x0000AFEF;
+
+	gpio->GPHCON = 0x0000FAAA;
+	gpio->GPHUP = 0x000007FF;
+
+#else
+#error Please define GTA01 version
+#endif
 
 	/* arch number of SMDK2410-Board */
 	gd->bd->bi_arch_number = MACH_TYPE_NEO1973;
@@ -144,5 +173,9 @@
 
 u_int32_t get_board_rev(void)
 {
+#if defined(CONFIG_ARCH_GTA01_v3)
 	return 0x00000130;
+#elif defined(CONFIG_ARCH_GTA01_v4)
+	return 0x00000140;
+#endif
 }
Index: u-boot.git/board/neo1973/split_by_variant.sh
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ u-boot.git/board/neo1973/split_by_variant.sh	2007-02-06 12:07:48.000000000 +0100
@@ -0,0 +1,36 @@
+#!/bin/sh
+# ---------------------------------------------------------
+#  Set the core module defines according to Core Module
+# ---------------------------------------------------------
+# ---------------------------------------------------------
+# Set up the GTA01 type define
+# ---------------------------------------------------------
+
+mkdir -p ${obj}include
+if [ "$1" == "" ]
+then
+	echo "$0:: No parameters - using GTA01v3 config"
+	echo "#define CONFIG_ARCH_GTA01_v3" > ${obj}include/config.h
+else
+	case "$1" in
+	gta01v4_config)
+	echo "#define CONFIG_ARCH_GTA01_v4" > ${obj}include/config.h
+	;;
+
+	gta01v3_config)
+	echo "#define CONFIG_ARCH_GTA01_v3" > ${obj}include/config.h
+	;;
+
+
+	*)
+	echo "$0:: Unrecognised config - using gta01v3_config"
+	echo "#define CONFIG_ARCH_GTA01_v3" > ${obj}include/config.h
+	;;
+
+	esac
+
+fi
+# ---------------------------------------------------------
+# Complete the configuration
+# ---------------------------------------------------------
+$MKCONFIG -a neo1973 arm arm920t neo1973 NULL s3c24x0
Index: u-boot.git/include/configs/neo1973.h
===================================================================
--- u-boot.git.orig/include/configs/neo1973.h	2007-02-06 12:04:53.000000000 +0100
+++ u-boot.git/include/configs/neo1973.h	2007-02-06 12:07:48.000000000 +0100
@@ -131,7 +131,11 @@
  * Miscellaneous configurable options
  */
 #define	CFG_LONGHELP				/* undef to save memory		*/
+#if defined(CONFIG_ARCH_GTA01_v3)
 #define	CFG_PROMPT		"GTA01v3 # "	/* Monitor Command Prompt	*/
+#elif defined(CONFIG_ARCH_GTA01_v4)
+#define	CFG_PROMPT		"GTA01v4 # "	/* Monitor Command Prompt	*/
+#endif
 #define	CFG_CBSIZE		256		/* Console I/O Buffer Size	*/
 #define	CFG_PBSIZE (CFG_CBSIZE+sizeof(CFG_PROMPT)+16) /* Print Buffer Size */
 #define	CFG_MAXARGS		16		/* max number of command args	*/
